rs/ic_os/hostos_tool/src/main.rs

```diff
@@ -62,26 +62,19 @@ pub fn main() -> Result<()> {
             eprintln!("Using config: {:?}", config_map);
 
             let network_info = NetworkInfo::from_config_map(&config_map)?;
-            eprintln!("Network info config: {:?}", &network_info);
 
-            let deployment = read_deployment_file(Path::new(&opts.deployment_file));
+            let network_settings = network_info.to_network_settings();
+            eprintln!("Network settings config: {:?}", &network_settings);
 
-            let deployment_name: Option<&str> = match &deployment {
-                Ok(deployment) => Some(deployment.deployment.name.as_str()),
-                Err(e) => {
-                    eprintln!("Error retrieving deployment file: {e}. Continuing without it");
-                    None
-                }
-            };
+            let deployment = read_deployment_file(Path::new(&opts.deployment_file));
 
-            let mgmt_mac: Option<&str> = match &deployment {
-                Ok(deployment) => deployment.deployment.mgmt_mac.as_deref(),
-                Err(_) => None,
-            };
+            let deployment_name = match &deployment {
+                Ok(deployment) => Ok(deployment.deployment.name.as_str()),
+                Err(e) => Err(anyhow::format_err!("Error retrieving deployment file: {e}")),
+            }?;
```

Comment:
Before, generate_network_config took a deployment_name Option and then errored if it was none. Now, generate_network_config takes a &str for deployment_name, and instead we error here if deployment_name is not found, so functionally this should be no different. 

The only reason I made this change here is because deployment_name is stored as a &str in the config struct, and I didn't want to turn it back into an option when calling generate_network_config in setupos_tool/main.rs